telemetry/definitions/commonDefinitions.json

```diff
@@ -1297,6 +1297,11 @@
             "type": "boolean",
             "description": "Whether the user has access to CodeWhisperer Chat"
         },
+        {
+            "name": "hasLogEventFilterPattern",
```

Comment:
This is over-specific. Can you find a field name that will be useful for other metrics in the future, e.g. `hasPatternFilter` (why is "log event" important enough to call out in the field name?)



Another way of future-proofing is to use enum-like (or string) fields instead of booleans. It's pretty cumbersome that we have booleans for each of these variants, and may need to add even more in the future.

Reply:
yes I missed _hasTextFilter_ which in itself is over specified and could have been _filterType_ with enum of text, time, log event etc. but at times metrics requires multiple fields each with it's corresponding boolean values. such has CW logs could have both time & text filter and hence we would go with two fields. 

In this case, I agree we could reuse _hasTextFilter_. @keeganirby could you please reuse _hasTextFilter_ in codebase and then we can safely remove _hasLogEventFilterPattern_

Reply:
Yeah we can use hasTextFilter.